blotztask-api/Services/TaskService.cs

```diff
@@ -137,12 +137,12 @@ public async Task<int> CompleteTask(int taskId)
             throw new NotFoundException($"Task with ID {taskId} was not found.");
         }
 
-        task.IsDone = true;
-
+        task.IsDone = !task.IsDone;
```

Comment:
this logic is good for now , but we can improve this further more 

```suggestion
    // Explicitly set or toggle the state
    task.IsDone = isDone ?? !task.IsDone;
```

this will then support for explicitly set the status as well using parameter

Reply:
Done it with task.isDone
`task.IsDone = (task.IsDone == null) ? false : !task.IsDone;`

Reply:
Fixed it using `task.IsDone = isDone ?? !task.IsDone;`